src/librustc/diagnostics.rs

```diff
@@ -1431,6 +1431,21 @@ fn make_child<'elve>(x: &mut &'elve isize, y: &mut &'elve isize) {
 ```
 "##,
 
+E0317: r##"
+An `if` expression without an `else` block is required to have the type `()`.
+This error occurs when the `if` block has a type other than `()`. For example:
+
+```compile_fail,E0317
+fn main() {
+    let x = 5;
+    let a = if x == 5 { 1 };
```

Comment:
This example can be made more relatable if written as such:

```
if condition {
    some_value
}
```

Missing semicolon after last expression in an `if` is likely the most common reason for this error by a large degree.